### ISSUE TITLE
Tracking gcc fhardened

### ISSUE THREAD

**Issue by Et7f3** [NixOS/nixpkgs#255433]
https://www.phoronix.com/news/GCC-fhardened-Hardening-Option

I don't see mention about fhardened in nixpkgs so we might want to enable it ?

i know we have hardening in place but IIRC we don't have -D_GLIBCXX_ASSERTIONS nor cf-protection set covered by this option

**Comment by trofi** [comment]
`-fhardened` is not yet added to upstream `gcc`.

**Comment by Et7f3** [comment]
Yes but we can do the preliminary works of testing which package will be broken with the equivalent flags.

Anyways I just posted here so we don't forget about it

**Comment by ethancedwards8** [comment]
Any status on this?
